src/main/java/it/gov/pagopa/arc/service/CustomAuthenticationSuccessHandler.java

```diff
@@ -35,13 +40,35 @@ public class CustomAuthenticationSuccessHandler implements AuthenticationSuccess
   @Override
   public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
       Authentication authentication) throws IOException {
+
     OAuth2AuthenticationToken oauth2Token = (OAuth2AuthenticationToken) authentication;
+    IamUserInfoDTO userInfoDTO = IamUserInfoDTO.map2IamUserInfoDTO( oauth2Token.getPrincipal().getAttributes());
+
+    if( isInWhiteList(userInfoDTO.getFiscalCode()) ){
+      TokenResponse accessToken = new TokenResponse(
+          accessTokenBuilderService.build(),
+          jwtConfiguration.getTokenType(),
+          jwtConfiguration.getAccessToken().getExpireIn(),
+          null,
+          null);
+
+      tokenStoreService.save( accessToken.getAccessToken() , userInfoDTO );
+
+      response.setContentType(MediaType.APPLICATION_JSON_VALUE);
+      response.getWriter().write( objectMapper.writeValueAsString(accessToken) );
+      response.getWriter().flush();
+    } else {
+      authentication.setAuthenticated(false);
+      response.setStatus(403);
+      response.setContentType(MediaType.APPLICATION_JSON_VALUE);
+      response.getWriter().write("{\"error\": \"User not allowed to access this application\"}");
+      response.getWriter().flush();
+    }
+
```

Comment:
as you can see, you are repeating some code! let's make it more simples
```suggestion
      body=  objectMapper.writeValueAsString(accessToken);
    } else {
      authentication.setAuthenticated(false);
      response.setStatus(403);
      body="{\"error\": \"User not allowed to access this application\"}";
    }
    
      response.setContentType(MediaType.APPLICATION_JSON_VALUE);
      response.getWriter().write( body );
      response.getWriter().flush();

```